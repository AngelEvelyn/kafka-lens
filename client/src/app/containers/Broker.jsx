import React, { Component } from 'react';
import BrokerView from '../components/BrokerView.jsx';

class Broker extends Component {
  constructor(props) {
    super(props);
    this.state = {
      brokers: [
        {
          brokerId: 1,
          brokerURI: 'localhost:test',
          isAlive: true,
          topics: [{ topicName: 'testName', newMessagesPerSecond: 15 }]
        },
        {
          brokerId: 2,
          brokerURI: 'localhost:test2',
          isAlive: false,
          topics: [
            { topicName: 'testTopic1', newMessagesPerSecond: 15 },
            { topicName: 'testTopic2', newMessagesPerSecond: 2340 }
          ]
        }
      ]
    };
  }

  /*
  // create method to parse through our data from the backend and set our new state
  componentDidMount() {
    // channel of listner to retain broker information from the backend and set state
    ipcRender.on('', (e, data) => {
      this.setState({
        brokers: data
      });
    });
  }
  */

  render() {
    const arr = [];
    for (let i = 0; i < this.state.brokers.length; i += 1) {
      arr.push(<BrokerView key={i} {...this.state.brokers[i]} />);
    }
    return <div>{arr}</div>;
  }
}

export default Broker;
