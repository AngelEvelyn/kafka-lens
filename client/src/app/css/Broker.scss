@import 'theme';

.broker-grid-container {
  min-height: 100vh;
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(320px, 350px));
  column-gap: 0.5em;
  row-gap: 0.5em;
  align-items: start;
  border-color: black;
  border-style: solid;
  overflow-y: scroll;
}

p {
  font-size: 20px;
}

.broker-header {
  text-align: center;
  font-size: 20px;
  font-weight: bold;
}
