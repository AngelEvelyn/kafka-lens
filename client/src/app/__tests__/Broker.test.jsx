import React from 'react';
import { shallow } from 'enzyme';

import Broker from '../containers/Broker';

describe('Broker.jsx unit tests', () => {
  let wrapper;
  const props = {
    kafkaHostURI: 'localhost:9092',
    brokersSnapshots: [{ isAlive: true, brokerId: 5, brokerURI: 'localhost:9092', topics: [] }],
  };

  beforeAll(() => {
    wrapper = shallow(<Broker {...props} />);
  });

  it('Should render without errors', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render <BrokerViews/>', () => {
    const instance = wrapper.instance();
    expect(instance.state.brokersSnapshots).toEqual([
      { isAlive: true, brokerId: 5, brokerURI: 'localhost:9092', topics: [] },
    ]);
    wrapper.setState({
      brokersSnapshots: [
        [
          { isAlive: true, brokerId: 1, brokerURI: 'localhost:9092', topics: [] },
          { isAlive: true, brokerId: 2, brokerURI: 'localhost:9092', topics: [] },
          { isAlive: false, brokerId: 3, brokerURI: 'localhost:9092', topics: [] },
        ],
      ],
    });
    expect(wrapper.exists('BrokerView')).toEqual(true);
  });
});
