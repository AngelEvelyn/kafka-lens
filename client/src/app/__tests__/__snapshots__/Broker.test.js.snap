// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Broker.js unit tests Should render without errors 1`] = `
<div
  className="broker-grid-container"
/>
`;
